packages/matter.js/src/behavior/definitions/descriptor/DescriptorServer.ts

```diff
@@ -48,23 +52,76 @@ export class DescriptorServer extends DescriptorBehavior {
                     revision: partType.deviceRevision,
                 },
             ];
+            for (
+                let base = MatterModel.standard.get(DeviceTypeModel, partType.deviceType)?.base;
+                base;
+                { base } = base
+            ) {
+                if (!(base instanceof DeviceTypeModel) || base.id === undefined) {
+                    continue;
+                }
+
+                this.state.deviceTypeList.push({
+                    deviceType: DeviceTypeId(base.id),
+                    revision: base.revision,
+                });
+            }
         }
     }
 
     /**
      * Extend device type metadata.  This is a shortcut for deduped insert into the deviceTypeList cluster attribute.
+     *
+     * @param deviceTypes an array of objects or named device types as defined in {@link MatterModel.standard}
      */
-    addDeviceTypes(...deviceTypes: DescriptorServer.DeviceType[]) {
+    addDeviceTypes(...deviceTypes: (DescriptorServer.DeviceType | string)[]) {
         const list = this.state.deviceTypeList;
-        nextInput: for (const newDeviceType of deviceTypes) {
-            for (const existingDeviceType of this.state.deviceTypeList) {
+
+        nextInput: for (let newDeviceType of deviceTypes) {
+            if (typeof newDeviceType === "string") {
+                const dt = MatterModel.standard.get(DeviceTypeModel, newDeviceType);
+                if (dt === undefined) {
+                    throw new ImplementationError(`Device type ${newDeviceType} not found`);
+                }
+                newDeviceType = { deviceType: DeviceTypeId(dt.id), revision: dt.revision };
+            }
+
+            for (const existingDeviceType of list) {
                 if (isDeepEqual(newDeviceType, existingDeviceType)) {
                     continue nextInput;
                 }
             }
             list.push(newDeviceType);
         }
-        this.state.deviceTypeList = list;
+    }
+
+    /**
+     * Add semantic tags.  This is a shortcut for deduped insert into the tagList cluster attribute.
+     */
```

Comment:
Please add here the info that for this method to work the feature needs to be enabled, so that devs see it at least in the IDE if they look

Reply:
OK